From 1acd15ca2ff92e259b2197cd72f4b49311256bec Mon Sep 17 00:00:00 2001
From: Arnaud Ferraris <arnaud.ferraris@collabora.com>
Date: Tue, 27 Aug 2024 10:16:07 +0200
Subject: [PATCH] src: gnss: fix build on 32-bit architectures

It seems `off_t` is now 64-bits wide on 32-bit architectures, so using
`glong` causes a build failure in this case. Switch to using the correct
type from the `sendfile()` prototype to fix this.
---
 src/gnss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gnss.c b/src/gnss.c
index 6503943..a43827a 100644
--- a/src/gnss.c
+++ b/src/gnss.c
@@ -327,7 +327,7 @@ static void init_assistance_data_upload(struct EG25Manager *manager)
 static void upload_assistance_data(struct EG25Manager *manager)
 {
     gint error;
-    glong written_total = 0;
+    off_t written_total = 0;
     gint ret;
     struct stat sb;
 
-- 
2.47.0

